android/src/main/java/com/bradbumbalough/RCTTwilioChat/RCTTwilioChatClient.java

```diff
@@ -294,7 +294,16 @@ public void onConnectionStateChange(ChatClient.ConnectionState state) {
 
     @Override
     public void onChannelAdd(final Channel channel) {
-        sendEvent("chatClient:channelAdded", RCTConvert.Channel(channel));
+        if (RCTTwilioChatClient.getInstance().client.getProperties().getSynchronizationStrategy() == ChatClient.SynchronizationStrategy.ALL) {
```

Comment:
I am wondering, why this is not happening on iOS SDK? I would like to submit support ticket to understand difference here.